resources/scripts/components/server/features/JavaVersionModalFeature.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useEffect, useState } from 'react';
+import { ServerContext } from '@/state/server';
+import Modal from '@/components/elements/Modal';
+import tw from 'twin.macro';
+import Button from '@/components/elements/Button';
+import setSelectedDockerImage from '@/api/server/setSelectedDockerImage';
+import FlashMessageRender from '@/components/FlashMessageRender';
+import useFlash from '@/plugins/useFlash';
+import { SocketEvent, SocketRequest } from '@/components/server/events';
+import Select from '@/components/elements/Select';
+
+const JavaVersionModalFeature = () => {
+    const [ visible, setVisible ] = useState(false);
+    const [ loading, setLoading ] = useState(false);
+
+    const uuid = ServerContext.useStoreState(state => state.server.data!.uuid);
+    const status = ServerContext.useStoreState(state => state.status.value);
+    const { clearFlashes, clearAndAddHttpError } = useFlash();
+    const { connected, instance } = ServerContext.useStoreState(state => state.socket);
+    let selectedVersion = 'ghcr.io/pterodactyl/yolks:java_16';
+
+    const dockerImageList = [
+        { name: 'Java 8', image: 'ghcr.io/pterodactyl/yolks:java_8' },
+        { name: 'Java 9', image: 'ghcr.io/pterodactyl/yolks:java_9' },
+        { name: 'Java 11', image: 'ghcr.io/pterodactyl/yolks:java_11' },
+        { name: 'Java 14', image: 'ghcr.io/pterodactyl/yolks:java_14' },
+        { name: 'Java 16', image: 'ghcr.io/pterodactyl/yolks:java_16' },
+    ];
+
+    useEffect(() => {
+        if (!connected || !instance || status === 'running') return;
+
+        const listener = (line: string) => {
+            if (line.toLowerCase().indexOf('minecraft 1.17 requires running the server with java 16 or above') >= 0) {
+                // Paper
+                setVisible(true);
+            } else if (line.toLowerCase().indexOf('java.lang.UnsupportedClassVersionError') >= 0) {
+                // Vanilla + many others
+                setVisible(true);
+            } else if (line.toLowerCase().indexOf('unsupported major.minor version') >= 0) {
+                // Forge
+                setVisible(true);
+            } else if (line.toLowerCase().indexOf('has been compiled by a more recent version of the java runtime') >= 0) {
+                // Everything else?
+                setVisible(true);
+            }
+        };
```

Comment:
This part could be DRY-ed up. And string prototype includes a beautiful "includes" method ;)

Reply:
I was just following what dane used for the eula feature. I'll look into it over the weekend for an improvement